speechbrain/decoders/seq2seq.py

```diff
@@ -1400,14 +1400,11 @@ class S2SWhisperBeamSearch(S2SBeamSearcher):
     """
 
     def __init__(
-        self, module, temperature=1.0, temperature_lm=1.0, **kwargs,
+        self, model, temperature=1.0, temperature_lm=1.0, **kwargs,
     ):
         super(S2SWhisperBeamSearch, self).__init__(**kwargs)
 
-        self.model = module[0]
```

Comment:
Is there a reason to drop CTC support?

Reply:
Just a point of view reason: I don't see why one would like to fine-tune using CTC as the Transformer decoder already is pre-trained - i.e. knows how to align properly. + for ST it's not good. If you can think of a reason to keep it, we'll keep it.

Reply:
> I don't see why one would like to fine-tune using CTC as the Transformer decoder already is pre-trained 

I am meaning that we can use CTC loss as an.auxiliary.loss during fine-tuning.

The reason to keep the CTC branch is that we can discard the transformer decoder and use only the CTC branch for decoding. Decoding with the attention decoder is rather slow according to past experience.

If someone does not want to use the CTC branch, he/she can ignore it and uses the transformer decoder for decoding.

Reply:
> The reason to keep the CTC branch is that we can discard the transformer decoder and use only the CTC branch for decoding. Decoding with the attention decoder is rather slow according to past experience.

So you mean using only the encoder + a different decoder than the whisper decoder right?

Reply:
I am meaning that we only use the transformer encoder + the CTC branch (i.e. ctc_linear + softmax) for decoding.

Reply:
@Adel-Moumen this makes sense, we should put back CTC xD

Reply:
> @Adel-Moumen this makes sense, we should put back CTC xD

Yes, that is only a small change but it provides us an alternative to use CTC for decoding.
I have not found any open-source project that is using CTC loss as an auxiliary loss when fine-tuning the whisper model.

Reply:
Well, we have a recipe that fine-tune the whisper encoder with CTC only in this PR as well ... but not both :p Let's add it again.

Reply:
The boss has spoken. I"ll put back the CTC decoding :p